voter-dapp/src/ActiveRequests.js

```diff
@@ -117,6 +122,23 @@ function ActiveRequests() {
   }, [subsequentFetchComplete, voteStatusesStringified, decryptionKeys, account]);
   const decryptionComplete = decryptedCommits && voteStatuses && decryptedCommits.length === voteStatuses.length;
 
+  const { send: batchRevealFunction, status } = useCacheSend("Voting", "batchReveal");
+  const onClickHandler = () => {
+    const reveals = [];
+    for (const index of Object.keys(checkboxesChecked)) {
+      if (checkboxesChecked[index]) {
+        reveals.push({
+          identifier: pendingRequests[index].identifier,
+          time: pendingRequests[index].time,
+          price: decryptedCommits[index].price.toString(),
+          salt: web3.utils.hexToNumberString(decryptedCommits[index].salt)
+        });
+      }
+    }
+    batchRevealFunction(reveals);
+    setCheckboxesChecked({});
```

Comment:
Just curious, do we not want to freeze the checkboxes until the result comes back? If we do, I think we should do that in a later PR, just wanted to hear your thoughts.

Reply:
Yeah, that definitely seems better. Not immediately obvious how to do that though haha.

Reply:
I say we leave it as is and figure it out later. This also might be useful if the transaction fails for whatever reason and the user wants to try to resubmit with the same boxes checked.